vcd/resource_vcd_edgegateway.go

```diff
@@ -353,3 +379,34 @@ func updateLoadBalancer(d *schema.ResourceData, egw govcd.EdgeGateway) error {
 
 	return nil
 }
+
+// resourceVcdEdgeGatewayImport is responsible for importing the resource.
+// The following steps happen as part of import
+// 1. The user supplies `terraform import _resource_name_ _the_id_string_` command
+// 2. `_the_id_string_` contains a dot formatted path to resource as in the example below
+// 3. The functions splits the dot-formatted path and tries to lookup the object
+// 4. If the lookup succeeds it sets the ID field for `_resource_name_` resource in statefile
+// (the resource must be already defined in .tf config otherwise `terraform import` will complain)
+// 5. `terraform refresh` is being implicitly launched. The Read method looks up all other fields
+// based on the known ID of object.
+//
+// Example resource name (_resource_name_): vcd_edgegateway.my-edge-gateway
+// Example import path (_the_id_string_): org.vdc.my-edge-gw
+func resourceVcdEdgeGatewayImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	resourceURI := strings.Split(d.Id(), ".")
+	if len(resourceURI) != 3 {
+		return nil, fmt.Errorf("resource name must be specified as org.vdc.edge-gw")
+	}
+	orgName, vdcName, edgeName := resourceURI[0], resourceURI[1], resourceURI[2]
+
+	vcdClient := meta.(*VCDClient)
+	edgeGateway, err := vcdClient.GetEdgeGateway(orgName, vdcName, edgeName)
+	if err != nil {
+		return nil, fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	_ = d.Set("org", orgName)
+	_ = d.Set("vdc", vdcName)
+	d.SetId(edgeGateway.EdgeGateway.ID)
+	return []*schema.ResourceData{d}, nil
```

Comment:
I am kind'a get idea that we have practise to call read after import?

Reply:
See n.5 above: refresh is called automatically.
In fact, I should remove the call to read where we do is explicitly.